Match the OS-level dark/light theme on all platforms

tangledhelix opened the pull request:
On all supported platforms, when Guiguts theme is set to "Default," detect the OS-level light/dark mode and dynamically switch Guiguts to match the OS.

Discontinue use of the Mac/Windows-styled "Default" theme. "Default" now means to match the OS light or dark, using the themes "Light" or "Dark".

Reduced contrast for background, text, and input-line highlight slight in the default themes. All other colors remain as they were.

Promoted the Image Viewer "High Contrast" mode to be a global preference, now applied to both the text area as well as images. This means there are now four possible theme choices: Light, Dark, and high-contrast variations of both.

Preference `IMAGE_VIEWER_HI_CONTRAST` replaced with `HIGH_CONTRAST`. This preference has moved to the top section, no longer Image Viewer-specific.

(Unrelated to dark/light mode): Fixed a mypy error on Mac related to the `idlelib.redirector` import.

Fixes #785

windymilla reviewed (approved):
The "blueness" was a bit of a shock to me too, but I'm happy for the reasons you said for us to let it go to users and see whether this theme (with the hi-contrast option) is OK for people generally.

Just one suggestion: move the Hi Contrast checkbox in the prefs dialog up, so that it sits directly below the theme selection, since the two belong together really.

tangledhelix commented:
> The "blueness" was a bit of a shock to me too, but I'm happy for the reasons you said for us to let it go to users and see whether this theme (with the hi-contrast option) is OK for people generally.

It's been changed to a pure gray of a similar brightness based on this comment and also conversation with @srjfoo.

> Just one suggestion: move the Hi Contrast checkbox in the prefs dialog up, so that it sits directly below the theme selection, since the two belong together really.

Done; rebased, squashed, pushed.

windymilla commented:
@tangledhelix I don't see any change in blueness - it's possible we were talking at cross-purposes, or I've not ended up refreshing my local version of the branch properly. I meant that in dark mode, the background is quite blue. As I said, if you think it's better, I'm happy to give it a try, but I thought I should check that I wasn't missing something that you had intended to change to "a puer gray".

srjfoo commented:
> @tangledhelix I don't see any change in blueness - it's possible we were talking at cross-purposes, or I've not ended up refreshing my local version of the branch properly. I meant that in dark mode, the background is quite blue. As I said, if you think it's better, I'm happy to give it a try, but I thought I should check that I wasn't missing something that you had intended to change to "a pure gray".

What I was talking about was specifically removing the blue from the text, so that's all I looked at when I was playing around. If you think the background is too blue, I can look at that, too.

windymilla commented:
No, happy to go with that if that was intended - I misunderstood. If all the users say "too blue", we'll reconsider :)
